registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/AnonymousProfileServiceImpl.java

```diff
@@ -56,5 +102,150 @@ public void saveAnonymousProfile(String regId, String processStage, String profi
 					"AnonymousProfileServiceImpl::saveAnonymousProfile()::exit");
 
 	}
+	
+	@Override
+	public String buildJsonStringFromPacketInfo(BiometricRecord biometricRecord, Map<String, String> fieldMap,
+			Map<String, String> metaInfoMap, String statusCode, String processStage)
+			throws JSONException, ApisResourceAccessException, PacketManagerException, IOException {
+
+		AnonymousProfileDTO anonymousProfileDTO = new AnonymousProfileDTO();
+		anonymousProfileDTO.setProcessName(
+				getFieldValueFromMetaInfo(metaInfoMap, JsonConstant.METADATA, JsonConstant.REGISTRATIONTYPE));
+		anonymousProfileDTO.setProcessStage(processStage);
+		anonymousProfileDTO.setStatus(statusCode);
+		String date = metaInfoMap.get(JsonConstant.CREATIONDATE);
+		anonymousProfileDTO.setDate(DateUtils.getTimeFromDate(
+				DateUtils.parseToDate(date, ANONYMOUS_PROFILE_DATETIME_PATTERN), ANONYMOUS_PROFILE_DATETIME_PATTERN));
+
+		String dateArr[] = fieldMap.get("dateOfBirth").split("/");
+		anonymousProfileDTO.setYearOfBirth(dateArr[0]);

```

Comment:
anonymousProfileDTO.setYearOfBirth(dateArr[0]);
this might not be correct . can you get dob string and convert it to a date object and then read year

Reply:
corrected... first converted to date and reading the year...